### PR TITLE
FIX: Do not trigger post alerts for empty posts.

### PR DESCRIPTION
**Pull request by nbianca**

### CODE REVIEW

File: spec/components/post_creator_spec.rb [hunk 1]
```diff
@@ -1141,6 +1141,20 @@
...
+      SiteSetting.queue_jobs = false
+      user2 = Fabricate(:user)
+      topic = Fabricate(:private_message_topic,
+        topic_allowed_users: [
+          Fabricate.build(:topic_allowed_user, user: user),
+          Fabricate.build(:topic_allowed_user, user: user2)
+        ],
+      )
+
+      PostAlerter.any_instance.expects(:after_save_post).never
```
Comment (tgxworld) [inline spec/components/post_creator_spec.rb]: Is there another way to test this? We generally do not want to use stubs or mocks in our test suite.

File: spec/components/post_creator_spec.rb [hunk 2]
```diff
@@ -1141,6 +1141,20 @@
...
+      PostCreator.create(user, raw: "", topic_id: topic.id, skip_validations: true)
```
Comment (tgxworld) [inline spec/components/post_creator_spec.rb]:
```suggestion
      PostCreator.create!(user, raw: "", topic_id: topic.id, skip_validations: true)
```

File: app/jobs/regular/post_alert.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ class PostAlert < Jobs::Base
 
     def execute(args)
       post = Post.find_by(id: args[:post_id])
-      if post&.topic
+      if post&.topic && !post.raw.blank?
```
Comment (tgxworld) [inline app/jobs/regular/post_alert.rb]: Can't really remember now but are post actions represented as post with empty raw?

Comment (ZogStriP) [inline app/jobs/regular/post_alert.rb]: It's better to always use the positive `post.present?` is easier to understand than `!post.blank?`

```suggestion
      if post&.topic && post.raw.present?
```

**Comment by SamSaffron** [comment]
The trick here is to create it for real with someone that would be notified normally and then dig into the notifications table.

File: spec/components/post_creator_spec.rb [hunk 4]
```diff
@@ -1141,6 +1141,26 @@
       post_creator = PostCreator.new(user, title: '', raw: '')
       expect { post_creator.create! }.to raise_error(ActiveRecord::RecordNotSaved)
     end
+
+    it "does not generate an alert for empty posts" do
+      SiteSetting.queue_jobs = false
```
Comment (SamSaffron) [inline spec/components/post_creator_spec.rb]: oh ... sorry ... but @eviltrout just killed this setting, can you migrate to the new pattern?

File: spec/components/post_creator_spec.rb [hunk 5]
```diff
@@ -1141,6 +1141,26 @@
...
+      )
+      Fabricate(:topic_user,
+        topic: topic,
+        user: user2,
+        notification_level: TopicUser.notification_levels[:watching]
+      )
+
+      expect {
+        PostCreator.create!(user, raw: "", topic_id: topic.id, skip_validations: true)
+      }.to change { user2.notifications.count }.by(0)
```
Comment (SamSaffron) [inline spec/components/post_creator_spec.rb]: I vaguely recall we may need to enable something else... to be extra safe here check that count changes by 1 and add

PostCreator.create!(user, raw: "hello world", topic_id: topic.id, skip_validations: true)

**Pull request merged**